adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -108,11 +108,30 @@ def __init__(
         if self._socknum == SOCKET_INVALID:
             raise RuntimeError("Failed to allocate socket.")
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.disconnect()
+        while self.status == adafruit_wiznet5k.SNSR_SOCK_FIN_WAIT:
+            pass
+        self.close()
+
     @property
     def socknum(self):
         """Returns the socket object's socket number."""
         return self._socknum
 
+    @socknum.setter
+    def socknum(self, socknum):
```

Comment:
Is this called anywhere else in the library aside from the new `accept()` call? It doesn't adhere to the CPython socket API (https://docs.python.org/3/library/socket.html#module-contents) and we may want to consider removing it and directly accusing `_socknum` within `accept`

Reply:
I originally referenced _socknum directly, though pylint flagged that. I can change back to direct references if you prefer.

Reply:
Yeah, please directly reference it since I want this socket to mirror the CPython `socket` for consistency . Does CPython expose a public `socknum` for a `socket` object?

Reply:
[It doesn't look like cpython has a socknum property](https://hg.python.org/cpython/file/tip/Lib/socket.py). I'll make the switch back to the direct references and roll it in with the other changes.